src/main/java/nl/tudelft/lifetiles/bucket/model/BucketCacheFactory.java

```diff
@@ -0,0 +1,28 @@
+package nl.tudelft.lifetiles.bucket.model;
+
+import nl.tudelft.lifetiles.graph.model.Graph;
+import nl.tudelft.lifetiles.sequence.model.SequenceSegment;
+
+/**
+ * A factory to produce a bucket cache.
+ *
+ * @author Jos
+ *
+ */
+public class BucketCacheFactory {
+
+    /**
+     * Constructs a bucket cache.
+     *
+     * @param n
+     *            Number of buckets to divide the graph in.
+     * @param graph
+     *            Graph to insert into the bucket cache.
+     * @return constructed bucket cache with cached graph in n buckets.
+     */
+    public final BucketCache getBucketCache(final int n,
```

Comment:
The factory pattern dictates that a factory should only return abstract types. But because this adds a lot of extra complexity, we decided that for simple models like `BucketCache` we shouldn't use factories (Although I would prefer a static factory method inside `BucketCache` itself).